### PR TITLE
Updates to match Node and Ember's LTS policies.

### PR DESCRIPTION
**Pull request by nathanhammond**
Specifies versions of Node.js per the LTS WG. Incremental step toward making the Ember ecosystem able to support [engine-strict](https://docs.npmjs.com/misc/config#engine-strict).

(PRs like these will show up when Greenkeeper asks us to bump versions of deps.)

### CODE REVIEW

**Comment by stefanpenner** [comment]
Until there is a hard blocker, I think this project should continue to support as many nodes as possible. (even unsupported ones). Remember this project actually supports many many others environments, from Titanium, vertx etc..

Thoughts?

**Comment by nathanhammond** [comment]
I have a preference for making EOL mean it so that people expect support to end and make plans to move off of old versions of libraries.

People continue to use old things *because we enable them*; it's addictive behavior. We should instead start saying "use an old unsupported version of the library."

Note that we've already stopped testing against 0.10 because one of our devDeps (Rollup) doesn't support it. If this PR is rejected I want to encourage us to fix that. Either support it and test against it, or drop support.

**Comment by stefanpenner** [comment]
> Either support it and test against it, or drop support.

Maybe it should be fixed, i'll have to think about it.

@nathanhammond my thoughts on this are:

Certain "infrastructure pieces" such as es6-promise and RSVP etc, should attempt (if reasonable) to support as many platforms as possible. If it becomes unreasonable to support something unsupported then we should absolutely drop it, and do a major version bump.

**Comment by nathanhammond** [comment]
Also then we should add tests for vertx, Titanium, Rhino, and anything else we theoretically support. If it's untested then we don't _truly_ support it.

File: package.json [hunk 1]
```diff
@@ -57,5 +57,8 @@
     "futures"
   ],
   "author": "Tilde, Inc. & Stefan Penner",
-  "license": "MIT"
+  "license": "MIT",
+  "engines": {
+    "node": "0.12.* || ^4.2 || 6.*"
```
Comment (stefanpenner) [inline package.json]: does this result in a warning fro NPM?

File: .travis.yml [hunk 2]
```diff
@@ -1,7 +1,6 @@
 language: node_js
 node_js:
   - "6"
-  - "5"
```
Comment (stefanpenner) [inline .travis.yml]: superseded by: https://github.com/tildeio/rsvp.js/pull/443

File: package.json (hunk 1 above)
Comment (nathanhammond) [inline package.json]: `npm` won't actually warn for dependencies. But we should carry it forward for `yarn` which will respect it.

`"0.12.* || 4.* || 6.* || 7.*"` is what we've started doing, though what we really mean is "latest of each branch."

**Comment by nathanhammond** [comment]
Rebased, good to land.

**Comment by stefanpenner** [comment]
> Also then we should add tests for vertx, Titanium, Rhino, and anything else we theoretically support. If it's untested then we don't truly support it.

Ya, maybe we don't "truly" support. But we "best effort" support. Some of those things are a tad tricky to get running on CI. Maybe someday

**Pull request merged**